### PR TITLE
metricbeat: collect total thread count per process

### PR DESCRIPTION
**Pull request by AndersonQ**
Collect the total by thread count per process upgrading elastic-agent-system-metrics to v0.7.0

## Proposed commit message

metricbeat: collect total thread count per process

Upgrades elastic-agent-system-metrics to v0.5.0 in order to collect the total thread count per process.

## Checklist

- [x] My code follows the style guidelines of this project
- ~~[ ] I have commented my code, particularly in hard-to-understand areas~~
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- ~~[ ] I have added tests that prove my fix is effective or that my feature works~~
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

- Depends on https://github.com/elastic/elastic-agent-system-metrics/pull/100
- Closes https://github.com/elastic/beats/issues/34461

## Logs

```
❯ go test -v -run TestFetch  ./metricbeat/module/system/process
=== RUN   TestFetch

// debug logs omitted
 
    process_test.go:51: fetched 5 events, showing events[0]:
    process_test.go:52: system/process event: {
          "process": {
            "args": [
              "/tmp/go-build658655655/b001/process.test",
              "-test.testlogfile=/tmp/go-build658655655/b001/testlog.txt",
              "-test.paniconexit0",
              "-test.timeout=10m0s",
              "-test.v=true",
              "-test.run=TestFetch"
            ],
            "command_line": "/tmp/go-build658655655/b001/process.test -test.testlogfile=/tmp/go-build658655655/b001/testlog.txt -test.paniconexit0 -test.timeout=10m0s -test.v=true -test.run=TestFetch",
            "cpu": {
              "pct": 0.0171,
              "start_time": "2023-09-13T11:52:45.000Z"
            },
            "executable": "/tmp/go-build658655655/b001/process.test",
            "memory": {
              "pct": 0.0004
            },
            "name": "process.test",
            "parent": {
              "pid": 196866
            },
            "pgid": 196866,
            "pid": 197122,
            "state": "sleeping",
            "working_directory": "/home/ainsoph/devel/github.com/elastic/beats/metricbeat/module/system/process"
          },
          "service": {
            "type": "system"
          },
          "system": {
            "process": {
              "cgroup": {
                "cgroups_version": 2,
                "cpu": {
                  "id": "vte-spawn-fa31675c-a919-4c0a-9447-9fdc56de8544.scope",
                  "path": "/user.slice/user-1000.slice/user@1000.service/app.slice/vte-spawn-fa31675c-a919-4c0a-9447-9fdc56de8544.scope",
                  "pressure": {
                    "full": {
                      "10": {
                        "pct": 0
                      },
                      "300": {
                        "pct": 0
                      },
                      "60": {
                        "pct": 0
                      },
                      "total": 4383214
                    },
                    "some": {
                      "10": {
                        "pct": 0
                      },
                      "300": {
                        "pct": 0
                      },
                      "60": {
                        "pct": 0
                      },
                      "total": 4588411
                    }
                  },
                  "stats": {
                    "system": {
                      "norm": {
                        "pct": 0
                      },
                      "ns": 30786963,
                      "pct": 0.0002
                    },
                    "usage": {
                      "norm": {
                        "pct": 0
                      },
                      "ns": 54487651,
                      "pct": 0.0003
                    },
                    "user": {
                      "norm": {
                        "pct": 0
                      },
                      "ns": 23700688,
                      "pct": 0.0001
                    }
                  }
                },
                "id": "vte-spawn-fa31675c-a919-4c0a-9447-9fdc56de8544.scope",
                "memory": {
                  "id": "vte-spawn-fa31675c-a919-4c0a-9447-9fdc56de8544.scope",
                  "mem": {
                    "events": {
                      "high": 0,
                      "low": 0,
                      "max": 0,
                      "oom": 0,
                      "oom_kill": 0
                    },
                    "low": {
                      "bytes": 0
                    },
                    "usage": {
                      "bytes": 551112704
                    }
                  },
                  "memsw": {
                    "events": {
                      "fail": 0,
                      "high": 0,
                      "max": 0
                    },
                    "low": {
                      "bytes": 0
                    },
                    "usage": {
                      "bytes": 0
                    }
                  },
                  "path": "/user.slice/user-1000.slice/user@1000.service/app.slice/vte-spawn-fa31675c-a919-4c0a-9447-9fdc56de8544.scope",
                  "stats": {
                    "active_anon": {
                      "bytes": 74604544
                    },
                    "active_file": {
                      "bytes": 161099776
                    },
                    "anon": {
                      "bytes": 64045056
                    },
                    "anon_thp": {
                      "bytes": 14680064
                    },
                    "file": {
                      "bytes": 400678912
                    },
                    "file_dirty": {
                      "bytes": 34336768
                    },
                    "file_mapped": {
                      "bytes": 38699008
                    },
                    "file_thp": {
                      "bytes": 0
                    },
                    "file_writeback": {
                      "bytes": 0
                    },
                    "htp_collapse_alloc": 0,
                    "inactive_anon": {
                      "bytes": 24576
                    },
                    "inactive_file": {
                      "bytes": 239509504
                    },
                    "kernel_stack": {
                      "bytes": 1376256
                    },
                    "major_page_faults": 1666,
                    "page_activate": 0,
                    "page_deactivate": 0,
                    "page_faults": 987725,
                    "page_lazy_free": 0,
                    "page_lazy_freed": 0,
                    "page_refill": 0,
                    "page_scan": 0,
                    "page_steal": 0,
                    "page_tables": {
                      "bytes": 835584
                    },
                    "per_cpu": {
                      "bytes": 1728
                    },
                    "shmem": {
                      "bytes": 0
                    },
                    "shmem_thp": {
                      "bytes": 0
                    },
                    "slab": {
                      "bytes": 71222912
                    },
                    "slab_reclaimable": {
                      "bytes": 69878216
                    },
                    "slab_unreclaimable": {
                      "bytes": 1344696
                    },
                    "sock": {
                      "bytes": 0
                    },
                    "swap_cached": {
                      "bytes": 0
                    },
                    "thp_fault_alloc": 876,
                    "unevictable": {
                      "bytes": 0
                    },
                    "workingset_activate_anon": 0,
                    "workingset_activate_file": 0,
                    "workingset_node_reclaim": 0,
                    "workingset_refault_anon": 0,
                    "workingset_refault_file": 0,
                    "workingset_restore_anon": 0,
                    "workingset_restore_file": 0
                  }
                },
                "path": "/user.slice/user-1000.slice/user@1000.service/app.slice/vte-spawn-fa31675c-a919-4c0a-9447-9fdc56de8544.scope"
              },
              "cmdline": "/tmp/go-build658655655/b001/process.test -test.testlogfile=/tmp/go-build658655655/b001/testlog.txt -test.paniconexit0 -test.timeout=10m0s -test.v=true -test.run=TestFetch",
              "cpu": {
                "start_time": "2023-09-13T11:52:45.000Z",
                "system": {
                  "ticks": 930
                },
                "total": {
                  "norm": {
                    "pct": 0.0171
                  },
                  "pct": 0.2743,
                  "ticks": 1420,
                  "value": 1420
                },
                "user": {
                  "ticks": 490
                }
              },
              "fd": {
                "limit": {
                  "hard": 1048576,
                  "soft": 1048576
                },
                "open": 9
              },
              "memory": {
                "rss": {
                  "bytes": 29417472,
                  "pct": 0.0004
                },
                "share": 16908288,
                "size": 2642882560
              },
              "num_threads": 19,
              "state": "sleeping"
            }
          },
          "user": {
            "name": "ainsoph"
          }
        }
--- PASS: TestFetch (3.26s)
PASS
ok  	github.com/elastic/beats/v7/metricbeat/module/system/process	3.569s

```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -406,3 +406,5 @@ replace (
 
 // Exclude this version because the version has an invalid checksum.
 exclude github.com/docker/distribution v2.8.0+incompatible
```
Comment (AndersonQ) [inline go.mod:408]: As soon as https://github.com/elastic/elastic-agent-system-metrics/pull/96 gets merged and `elastic-agent-system-metrics` `v0.5.0` released, I'll remove the replace and properly upgrade `elastic-agent-system-metrics` to `v0.5.0`.

File: go.mod [hunk 2]
```diff
@@ -406,3 +406,5 @@ replace (
...
+
+replace github.com/elastic/elastic-agent-system-metrics => github.com/AndersonQ/elastic-agent-system-metrics v0.0.0-20230913114254-ef72b6fa59d9
```
Comment (AndersonQ) [inline go.mod:410]: As soon as https://github.com/elastic/elastic-agent-system-metrics/pull/96 gets merged and `elastic-agent-system-metrics` `v0.5.0` released, I'll remove the replace and properly upgrade `elastic-agent-system-metrics` to `v0.5.0`.

File: libbeat/tests/system/beat/beat.py [hunk 3]
```diff
@@ -677,6 +677,9 @@ def extract_fields(doc_list, name):
             if doc_list is None:
                 return fields, dictfields, aliases
 
+            # print("===================================================================================================")
+            # print("doc_list:", doc_list)
+            # print("===================================================================================================")
```
Comment (belimawr) [inline libbeat/tests/system/beat/beat.py:682]: Debug comments?
```suggestion
```

File: libbeat/tests/system/beat/beat.py [hunk 4]
```diff
@@ -717,6 +720,10 @@ def extract_fields(doc_list, name):
             if not os.path.isfile(path):
                 path = os.path.abspath(os.path.dirname(
                     __file__) + "../../../../_meta/fields.common.yml")
+
+                print(                    "=================================================================================================")
+                print("path:", path)
+                print(                    "=================================================================================================")
```
Comment (belimawr) [inline libbeat/tests/system/beat/beat.py:726]: Debug comments?
```suggestion
```

**Review by belimawr — DISMISSED** [review]

**Review by fearful-symmetry — APPROVED** [review]

**Review by jlind23 — APPROVED** [review]
Already approved by @fearful-symmetry

**Pull request merged**